client.py

```diff
@@ -32,6 +34,18 @@ def launch_glidein(cmd, params=[]):
 def get_running(cmd):
     p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
     return int(p.communicate()[0].strip())
+    
+def sort_states(state, columns, reverse=True):
+    comparers = [((itemgetter(col[1:].strip()), -1) if col.startswith('-') else
+                  (itemgetter(col.strip()), 1)) for col in columns]
+    def comparer(left, right):
+        for fn, mult in comparers:
+            result = cmp(fn(left), fn(right))
+            if result:
+                return mult * result
+        else:
+            return 0
+    return sorted(state, cmp=comparer, reverse=reverse)
```

Comment:
Instead of using `cmp`, sorted should use `key`.  Note the docs:

> In Python 3, the cmp parameter was removed entirely (as part of a larger effort to simplify and unify the language, eliminating the conflict between rich comparisons and the __cmp__() magic method).